src/octopeer-github/main/DatabaseAdapter.ts

```diff
@@ -0,0 +1,91 @@
+class DatabaseAdapter {
+
+    private _url: string;
+    private _session: number;
+    private _pr: number;
+    private _user: number;
+    private _initialized: boolean;
+    private _debug: boolean;
+
+    constructor(url: string, debug = false) {
+        const self = this; // scope 'self' to be the adapter (instead of the AJAX call)
+        self._debug = debug;
+        self._user = 1;
+        self._pr = 1;   // TODO The PR id should be get from the context and probably set via the constructor
+        self._session = -1;
+        self._initialized = false;
+
+        self._url = url;
+        if (self._url.charAt(this._url.length - 1) !== "/") {
+            self._url += "/";
+        }
+
+        $.ajax(`${self.url}api/sessions/`, {
+                data: {
+                    "platform": "GitHub",
+                    "pull_request": `${this.url}api/pull-requests/${this._pr}/`,
+                    "user": `${this.url}api/users/${this._user}/`,
+                },
+                dataType: "json",
+                type: "POST",
+            })
+            .done(function(data, status, jqXHR) {
+                let suburl = data.url.substr(0, data.url.length - 1); // trim trailing slash
+                suburl = suburl.substr(suburl.lastIndexOf("/") + 1);  // substring starting from the character after the last "/"
+                self._session = Number(suburl);
+                self._initialized = true;
+                if (this._debug) {
+                    console.log(`Initialized DatabaseAdapter(${url}, session=${self._session})`, self);
```

Comment:
If debug logging is this course, we should consider added a program-wide debug flag.